internal/services/webhooks/webhooks.go

```diff
@@ -76,88 +79,108 @@ func (c *WebhooksController) check() error {
-				tokenBytes, err := base64.StdEncoding.DecodeString(ww.TokenValue.String)
-				if err != nil {
-					c.sc.Logger.Error().Err(err).Msgf("failed to decode access token: %s", err.Error())
-					continue
-				}
-				decTok, err := c.sc.Encryption.Decrypt(tokenBytes, "engine_webhook_worker_token")
-				if err != nil {
-					c.sc.Logger.Error().Err(err).Msgf("failed to encrypt access token: %s", err.Error())
-					continue
-				}
+					encTok, err := c.sc.Encryption.Encrypt([]byte(tok.Token), "engine_webhook_worker_token")
+					if err != nil {
+						c.sc.Logger.Error().Err(err).Msgf("failed to encrypt access token: %s", err.Error())
+						return nil
+					}
 
-				token = string(decTok)
-			} else {
-				tok, err := c.sc.Auth.JWTManager.GenerateTenantToken(context.Background(), tenantId, "webhook-worker")
-				if err != nil {
-					c.sc.Logger.Error().Err(err).Msgf("could not generate token for webhook worker %s of tenant %s", id, tenantId)
-					continue
-				}
+					encTokStr := base64.StdEncoding.EncodeToString(encTok)
 
-				encTok, err := c.sc.Encryption.Encrypt([]byte(tok.Token), "engine_webhook_worker_token")
-				if err != nil {
-					c.sc.Logger.Error().Err(err).Msgf("failed to encrypt access token: %s", err.Error())
-					continue
-				}
+					_, err = c.sc.EngineRepository.WebhookWorker().UpsertWebhookWorker(context.Background(), &repository.UpsertWebhookWorkerOpts{
+						Name:       ww.Name,
+						URL:        ww.Url,
+						Secret:     ww.Secret,
+						TenantId:   tenantId,
+						TokenID:    &tok.TokenId,
+						TokenValue: &encTokStr,
+					})
+					if err != nil {
+						c.sc.Logger.Error().Err(err).Msgf("could not update webhook worker %s of tenant %s", id, tenantId)
+						return nil
+					}
 
-				encTokStr := base64.StdEncoding.EncodeToString(encTok)
+					token = tok.Token
+				}
 
-				_, err = c.sc.EngineRepository.WebhookWorker().UpsertWebhookWorker(context.Background(), &repository.UpsertWebhookWorkerOpts{
-					Name:       ww.Name,
-					URL:        ww.Url,
-					Secret:     ww.Secret,
-					TenantId:   tenantId,
-					TokenID:    &tok.TokenId,
-					TokenValue: &encTokStr,
-				})
+				cleanup, err := c.run(tenantId, ww, token, h)
 				if err != nil {
-					c.sc.Logger.Error().Err(err).Msgf("could not update webhook worker %s of tenant %s", id, tenantId)
-					continue
+					c.sc.Logger.Error().Err(err).Msgf("error running webhook worker %s of tenant %s healthcheck", id, tenantId)
+					return nil
+				}
+				if cleanup != nil {
+					c.cleanups[id] = cleanup
 				}
 
-				token = tok.Token
-			}
+				return nil
+			})
+		}
 
-			cleanup, err := c.run(tenantId, ww, token, h)
-			if err != nil {
-				c.sc.Logger.Error().Err(err).Msgf("error running webhook worker %s of tenant %s healthcheck", id, tenantId)
-				continue
-			}
-			if cleanup != nil {
-				c.cleanups = append(c.cleanups, cleanup)
-			}
+		if err = ec.Wait(); err != nil {
+			panic(err) // the error group above is not expected to error
```

Comment:
I'm still not sure we should panic here, or the entire engine will restart right? Unless there's somewhere you're handling panic/recover.

Reply:
this line will never happen, I could have written _  = ec.Wait() as well

Reply:
I will rewrite this

Reply:
done, it's just using a goroutine now